src/ophyd_async/panda/writers/_panda_hdf_file.py

```diff
@@ -30,7 +30,7 @@ def __init__(
                 spec="AD_HDF5_SWMR_SLICE",
                 root=str(directory_info.root),
                 data_key=ds.name,
-                resource_path=(f"{str(directory_info.root)}/{full_file_name}"),
+                resource_path=str(full_file_path.relative_to(directory_info.root)),
```

Comment:
A test is erroring here because `root` is absolute, however `full_file_path` is "`.`".